frappe/integrations/doctype/ldap_settings/test_ldap_settings.py

```diff
@@ -1,8 +1,1071 @@
+
+		# Create test user1
+		self.user1doc = {
+			'username': 'posix.user',
+			'email': 'posix.user1@unit.testing',
+			'first_name': 'posix'
+		}
+		self.user1doc.update({
+			"doctype": "User",
+			"send_welcome_email": 0,
+			"language": "",
+			"user_type": "System User",
+		})
+
+		user = frappe.get_doc(self.user1doc)
+		user.insert(ignore_permissions=True)
+
+		# Create test user1
+		self.user2doc = {
+			'username': 'posix.user2',
+			'email': 'posix.user2@unit.testing',
+			'first_name': 'posix'
+		}
+		self.user2doc.update({
+			"doctype": "User",
+			"send_welcome_email": 0,
+			"language": "",
+			"user_type": "System User",
+		})
+
+		user = frappe.get_doc(self.user2doc)
+		user.insert(ignore_permissions=True)
+
+
+		# Setup Mock OpenLDAP Directory
+		self.ldap_dc_path = 'dc=unit,dc=testing'
+		self.ldap_user_path = 'ou=users,' + self.ldap_dc_path
+		self.ldap_group_path = 'ou=groups,' + self.ldap_dc_path
+		self.base_dn = 'cn=base_dn_user,' + self.ldap_dc_path
+		self.base_password = 'my_password'
+		self.ldap_server = 'ldap://my_fake_server:389'
+
+
+		self.doc = {
+			"doctype": "LDAP Settings",
+			"enabled": True,
+			"ldap_directory_server": self.TEST_LDAP_SERVER,
+			"ldap_server_url": self.ldap_server,
+			"base_dn": self.base_dn,
+			"password": self.base_password,
+			"organizational_unit": self.ldap_user_path,
+			"ldap_search_string": self.TEST_LDAP_SEARCH_STRING,
+			"organizational_unit_for_groups": self.ldap_group_path,
+			"ldap_user_creation_and_mapping_section": '',
+			"ldap_email_field": 'mail',
+			"ldap_username_field": self.LDAP_USERNAME_FIELD,
+			"ldap_first_name_field": 'givenname',
+			"ldap_middle_name_field": '',
+			"ldap_last_name_field": 'sn',
+			"ldap_phone_field": 'telephonenumber',
+			"ldap_mobile_field": 'mobile',
+			"ldap_security": '',
+			"ssl_tls_mode": '',
+			"require_trusted_certificate": 'No',
+			"local_private_key_file": '',
+			"local_server_certificate_file": '',
+			"local_ca_certs_file": '',
+			"ldap_group_objectclass": '',
+			"ldap_group_member_attribute": '',
+			"default_role": 'Newsletter Manager',
+			"ldap_groups": self.DOCUMENT_GROUP_MAPPINGS,
+			"ldap_group_field": ''}
+
+		self.server = Server(host=self.ldap_server, port=389, get_info=self.LDAP_SCHEMA)
+
+		self.connection = Connection(
+			self.server,
+			user=self.base_dn,
+			password=self.base_password,
+			read_only=True,
+			client_strategy=MOCK_SYNC)
+
+		self.connection.strategy.entries_from_json(os.path.abspath( os.path.dirname( __file__ ) ) + '/' + self.LDAP_LDIF_JSON)
+
+		self.connection.bind()
+
+
+	@classmethod
+	def tearDownClass(self):
+		print(self.TEST_LDAP_SERVER + ' tear down')
+		try:
+			frappe.get_doc('LDAP Settings').delete()
+
+		except:
+			pass
+
+		try:
+			# return doc back to user data
+			self.user_ldap_settings.save()
+
+		except:
+			pass
+
+		# Clean-up test users
+		self.clean_test_users()
+
+		# Clear OpenLDAP connection
+		self.connection = None
+
+
+
+	def test_confirm_setup(self):
```

Comment:
refactor:

Remove. As this only tests the setup. If the setup fails, the dev working on will have exceptions to view to fault find.

Reply:
removed in 9f92fa02de69039990dd3b913df6a1da41681dbd